tests/JobDefinitionFactoryTest.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Tests;
+
+use App\JobDefinitionFactory;
+use Keboola\DockerBundle\Docker\Component;
+use Keboola\JobQueueInternalClient\JobFactory;
+use Keboola\JobQueueInternalClient\JobFactory\Job;
+use Keboola\ObjectEncryptor\ObjectEncryptor;
+use Keboola\ObjectEncryptor\ObjectEncryptorFactory;
+use Keboola\StorageApi\Client;
+use PHPUnit\Framework\TestCase;
+
+class JobDefinitionFactoryTest extends TestCase
+{
+    public function testCreateJobDefinitionWithConfigData(): void
+    {
+        $configData = [
+            'runtime' => [
+                'foo' => 'bar',
+            ],
+        ];
+
+        $jobData = [
+            'status' => JobFactory::STATUS_CREATED,
+            'projectId' => 'my-project',
+            'componentId' => 'my-component',
+            'configId' => 'my-config',
+            'configData' => $configData,
+        ];
+
+        $jobDefinitions = $this->createJobDefinitions($jobData, $configData);
+
+        self::assertCount(1, $jobDefinitions);
+        $jobDefinition = $jobDefinitions[0];
+
+        self::assertSame($jobData['configId'], $jobDefinition->getConfigId());
+        self::assertSame($jobData['componentId'], $jobDefinition->getComponentId());
+        self::assertSame('bar', $jobDefinition->getConfiguration()['runtime']['foo'] ?? null);
+    }
+
+    public function testCreateJobDefinitionWithConfigDataAndBackend(): void
+    {
+        $configData = [
+            'runtime' => [
+                'foo' => 'bar',
+                'backend' => [
+                    'type' => 'invalid',
+                ],
+            ],
+        ];
+
+        $jobData = [
+            'status' => JobFactory::STATUS_CREATED,
+            'projectId' => 'my-project',
+            'componentId' => 'my-component',
+            'configId' => 'my-config',
+            'configData' => $configData,
+            'backend' => [
+                'type' => 'custom',
+            ],
+        ];
+
+        $jobDefinitions = $this->createJobDefinitions($jobData, $configData);
```

Comment:
```suggestion
        $jobDefinitions = $this->createJobDefinitions($jobData, []);
```
protože v tomhle $configData <> $configuration

Reply:
Zkusil jsme tu hepler metody rozdelit na 2 - jedno pro testy s configData a jednu s configuration, mohlo by to ted snad byt prehlednejsi.